#!/usr/bin/env python
PACKAGE = "ueye"
import roslib;roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator import *

gen = ParameterGenerator()

gen.add("hor_mirror", bool_t, 0, "Horizontal Mirror", False)
gen.add("ver_mirror", bool_t, 0, "Vertical Mirror", False)

gen.add("auto_shutter", bool_t, 0, "Auto Shutter", True)
gen.add("auto_sensor_shutter", bool_t, 0, "Auto Sensor Shutter", False)
gen.add("auto_gain", bool_t, 0, "Auto Sensor Shutter", False)
gen.add("gain_boost", bool_t, 0, "Auto Sensor Shutter", False)
gen.add("pixel_clock", int_t, 0, "Pixel Clock (MHz)", 25 , 5, 40)

gen.add("framerate", double_t, 0, "Framerate (fps)", 30 , 1, 100)

gen.add("exposure_time", double_t, 0, "Exposure Time (ms)", 10, 1, 100)

gen.add("image_width", int_t, 0, "Width to be read on the sensor", 752 , 16, 752)
gen.add("image_height", int_t, 0, "Height to be read on the sensor", 480 , 4, 480)
gen.add("image_left_C1", int_t, 0, "Left Offset Camera 1", 0 , 0, 736)
gen.add("image_top_C1", int_t, 0, "Top offset Camera 1", 0 , 0, 476 )
gen.add("image_left_C2", int_t, 0, "Left Offset Camera 2", 0 , 0, 736)
gen.add("image_top_C2", int_t, 0, "Top offset Camera 2", 0 , 0, 476 )

exit(gen.generate(PACKAGE, "ueye", "params"))
